### ISSUE TITLE
[Bug]: qwen2.5-72b-insruct math 自测分数和榜单分数差异较大

### ISSUE THREAD

**Issue by tianshiyisi** [QwenLM/Qwen2.5#1020]
### Model Series

Qwen2.5

### What are the models used?

qwen2.5-72b-instruct

### What is the scenario where the problem happened?

qwen2.5-72b-instruct math评测与官方榜单差异较大

### Is this a known issue?

- [X] I have followed [the GitHub README](https://github.com/QwenLM/Qwen2.5).
- [X] I have checked [the Qwen documentation](https://qwen.readthedocs.io) and cannot find an answer there.
- [X] I have checked the documentation of the related framework and cannot find useful information.
- [X] I have searched [the issues](https://github.com/QwenLM/Qwen2.5/issues?q=is%3Aissue) and there is not a similar one.

### Information about environment

OS: Ubuntu 22.04
Python: Python 3.10.6
GPUs: 8 x NVIDIA A100
NVIDIA driver: 470.141.10 (from nvidia-smi)
CUDA compiler: 12.1 (from nvcc -V)
PyTorch: 2.4.0+cu121 (from python -c "import torch; print(torch.__version__)")

### Log output

```shell
压测过程日志：
10/15 16:01:01 - OpenCompass - INFO - Task [qwen2/math]
10/15 16:01:03 - OpenCompass - WARNING - Max Completion tokens for qwen2 is :16384
10/15 16:01:03 - OpenCompass - INFO - Try to load the data from /root/.cache/opencompass/./data/math/math.json
10/15 16:01:03 - OpenCompass - INFO - Start inferencing [qwen2/math]
[2024-10-15 16:01:05,267] [opencompass.openicl.icl_inferencer.icl_gen_inferencer] [INFO] Starting build dataloader
[2024-10-15 16:01:05,267] [opencompass.openicl.icl_inferencer.icl_gen_inferencer] [INFO] Starting inference process...
100%|██████████| 40/40 [58:08<00:00, 87.22s/it] 
10/15 16:59:14 - OpenCompass - INFO - time elapsed: 3492.95s

压测结果日志：
10/15 16:59:20 - OpenCompass - INFO - Try to load the data from /root/.cache/opencompass/./data/math/math.json
10/15 16:59:20 - OpenCompass - INFO - Task [qwen2/math]: {'accuracy': 49.18}
10/15 16:59:20 - OpenCompass - INFO - time elapsed: 2.93s
```

### Description

启动命令： python -m vllm.entrypoints.openai.api_server --port 8000 --tensor-parallel-size 4 --gpu-memory-utilization 0.95 --model /oss/Qwen2.5-72B-Instruct --served-model-name qwen2
压测命令：
cat <<'EOF' >eval_openai_api.json
{
  "eval_backend": "OpenCompass",
  "eval_config": {
    "datasets": [
      "math"
    ],
    "models": [
      {
        "path": "qwen2",
        "openai_api_base": "http://127.0.0.1:8000/v1/chat/completions",
        "temperature": 0.0
      }
    ]
  }
}
EOF

cat <<'EOF' >eval_openai_api.py
from evalscope.run import run_task
from evalscope.summarizer import Summarizer

def run_eval():

    task_cfg = 'eval_openai_api.json'

    run_task(task_cfg=task_cfg)

    print('>> Start to get the report with summarizer ...')
    report_list = Summarizer.get_report_from_cfg(task_cfg)
    print(f'\n>> The report list: {report_list}')

run_eval()
EOF

python eval_openai_api.py

压测结果（math 72b也测了一下）：
![image](https://github.com/user-attachments/assets/0292aa7f-3926-496b-bbc0-178520763c02)

**Comment by hzhwcmhf** [comment]
You can find the scripts for reproducing our results here: https://github.com/QwenLM/Qwen2.5-Math/tree/main?tab=readme-ov-file#evaluation

**Comment by tianshiyisi** [comment]
thanks @hzhwcmhf ,i will reproduce this and the score from
https://qwenlm.github.io/zh/blog/qwen2.5-llm/
![image](https://github.com/user-attachments/assets/1096a728-aebd-461b-8259-24702199c237)

**Comment by tianshiyisi** [comment]
The issue has been resolved, thanks. Using the Qwen2.5-math benchmarking tool to test on the math dataset, the Qwen2.5-72B-Instruct model scored 82.8, while the Qwen2.5-math-72B-Instruct model scored 85.3. The reason is that the prompts constructed by the two are different, specifically the position of the CoT (Chain of Thought) hint. The former prompt is more likely to lead to the correct answer.
Qwen2.5-math example:
curl -X POST \
    -H "Content-Type: application/json" \
    -d '{
        "model": "qwen2",
        "messages": [
        {
            "role": "system",
            "content": "You are a helpful assistant."
        },
        {
            "role": "user",
            "content": "A robot moving forward at a constant speed takes 2.5 hours to travel 1 kilometer. Moving forward at this same constant speed, it takes the robot 90 seconds to travel the length of a particular hallway. How many meters long is the hallway? Please reason step by step, and put your final answer within \\boxed{}."
        }
        ],
        "temperature": 0,
        "top_p": 1.0,
        "top_k": -1
}' http://127.0.0.1:8000/v1/chat/completions

OpenCompass prompt example:
curl -X POST \
    -H "Content-Type: application/json" \
    -d '{
        "model": "qwen2",
        "messages": [
        {
            "role": "system",
            "content": "Please reason step by step, and put your final answer within \\boxed{}."
        },
        {
            "role": "user",
            "content": "A robot moving forward at a constant speed takes 2.5 hours to travel 1 kilometer. Moving forward at this same constant speed, it takes the robot 90 seconds to travel the length of a particular hallway. How many meters long is the hallway?"
        }
        ],
        "temperature": 0,
        "top_p": 1.0,
        "top_k": -1
}' http://127.0.0.1:8000/v1/chat/completions

**Comment by lenohard** [comment]
I want to confirm if just the location of the CoT prompt contributes to a 30% accuracy boost

**Comment by tianshiyisi** [comment]
Yes, I verified that evalscope is using the math_gen_265cce.py file under opencompass. The content is as follows:
from opencompass.openicl.icl_prompt_template import PromptTemplate
from opencompass.openicl.icl_retriever import ZeroRetriever
from opencompass.openicl.icl_inferencer import GenInferencer
from opencompass.datasets import MATHDataset, MATHEvaluator, math_postprocess_v2

math_reader_cfg = dict(input_columns=['problem'], output_column='solution')

math_infer_cfg = dict(
    prompt_template=dict(
        type=PromptTemplate,
        template=dict(round=[
            dict(role='HUMAN', prompt='Problem:\nFind the domain of the expression $\\frac{{\sqrt{{x-2}}}}{{\sqrt{{5-x}}}}$.}}\nSolution:'),
            dict(role='BOT', prompt='The expressions inside each square root must be non-negative. Therefore, $x-2 \ge 0$, so $x\ge2$, and $5 - x \ge 0$, so $x \le 5$. Also, the denominator cannot be equal to zero, so $5-x>0$, which gives $x<5$. Therefore, the domain of the expression is $\\boxed{{[2,5)}}$.\nFinal Answer: The final answer is $[2,5)$. I hope it is correct.\n'),
            dict(role='HUMAN', prompt='Problem:\nIf $\det \mathbf{{A}} = 2$ and $\det \mathbf{{B}} = 12,$ then find $\det (\mathbf{{A}} \mathbf{{B}}).$\nSolution:'),
            dict(role='BOT', prompt='We have that $\det (\mathbf{{A}} \mathbf{{B}}) = (\det \mathbf{{A}})(\det \mathbf{{B}}) = (2)(12) = \\boxed{{24}}.$\nFinal Answer: The final answer is $24$. I hope it is correct.\n'),
            dict(role='HUMAN', prompt='Problem:\nTerrell usually lifts two 20-pound weights 12 times. If he uses two 15-pound weights instead, how many times must Terrell lift them in order to lift the same total weight?\nSolution:'),
            dict(role='BOT', prompt='If Terrell lifts two 20-pound weights 12 times, he lifts a total of $2\cdot 12\cdot20=480$ pounds of weight. If he lifts two 15-pound weights instead for $n$ times, he will lift a total of $2\cdot15\cdot n=30n$ pounds of weight. Equating this to 480 pounds, we can solve for $n$: \\begin{{align*}} 30n&=480\\\\ \Rightarrow\qquad n&=480/30=\\boxed{{16}} \end{{align*}}\nFinal Answer: The final answer is $16$. I hope it is correct.\n'),
            dict(role='HUMAN', prompt='Problem:\nIf the system of equations: \\begin{{align*}} 6x-4y&=a,\\\\ 6y-9x &=b. \end{{align*}}has a solution $(x, y)$ where $x$ and $y$ are both nonzero, find $\\frac{{a}}{{b}},$ assuming $b$ is nonzero.\nSolution:'),
            dict(role='BOT', prompt='If we multiply the first equation by $-\\frac{{3}}{{2}}$, we obtain $$6y-9x=-\\frac{{3}}{{2}}a.$$Since we also know that $6y-9x=b$, we have $$-\\frac{{3}}{{2}}a=b\Rightarrow\\frac{{a}}{{b}}=\\boxed{{-\\frac{{2}}{{3}}}}.$$\nFinal Answer: The final answer is $-\\frac{{2}}{{3}}$. I hope it is correct.\n'),
            dict(role='HUMAN', prompt='Problem:\n{problem}\nSolution:\n'),
        ])),
    retriever=dict(type=ZeroRetriever),
    inferencer=dict(type=GenInferencer, max_out_len=512))

math_eval_cfg = dict(
    evaluator=dict(type=MATHEvaluator, version='v2'), pred_postprocessor=dict(type=math_postprocess_v2))

math_datasets = [
    dict(
        type=MATHDataset,
        abbr='math',
        path='opencompass/math',
        reader_cfg=math_reader_cfg,
        infer_cfg=math_infer_cfg,
        eval_cfg=math_eval_cfg)
]
I modified it to use the math_0shot_gen_393424.py file with the following content:
from opencompass.openicl.icl_prompt_template import PromptTemplate
from opencompass.openicl.icl_retriever import ZeroRetriever
from opencompass.openicl.icl_inferencer import GenInferencer
from opencompass.datasets import MATHDataset, MATHEvaluator, math_postprocess_v2, normalize_final_answer

math_reader_cfg = dict(input_columns=['problem'], output_column='solution')

math_infer_cfg = dict(
    prompt_template=dict(
        type=PromptTemplate,
        template=dict(
            round=[
                dict(role='HUMAN', prompt='{problem}\nPlease reason step by step, and put your final answer within \\boxed{}.'),
            ]
        ),
    ),
    retriever=dict(type=ZeroRetriever),
    inferencer=dict(type=GenInferencer, max_out_len=1024),
)

# postprocess v2
math_eval_cfg = dict(
    evaluator=dict(type=MATHEvaluator, version='v2'), pred_postprocessor=dict(type=math_postprocess_v2),
)

math_datasets = [
    dict(
        type=MATHDataset,
        abbr='math',
        path='opencompass/math',
        reader_cfg=math_reader_cfg,
        infer_cfg=math_infer_cfg,
        eval_cfg=math_eval_cfg,
    )
]
After stress testing, the score improved from 49.18 to 75.18.

**Comment by 13416157913** [comment]
> Yes, I verified that evalscope is using the math_gen_265cce.py file under opencompass. The content is as follows: ![image](https://private-user-images.githubusercontent.com/76173133/377706213-927e1b16-d725-4b26-97de-668402c1868a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA5Njk1MzUsIm5iZiI6MTczMDk2OTIzNSwicGF0aCI6Ii83NjE3MzEzMy8zNzc3MDYyMTMtOTI3ZTFiMTYtZDcyNS00YjI2LTk3ZGUtNjY4NDAyYzE4NjhhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA3VDA4NDcxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlhYTU2YmVjYmE1MTViMGMyMzhmM2RjYTEwNjVkZTM3MTkyNjc3NzUyZmNlMTdiNWI3NzRhMGIwNmY2MDZjZTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ojBjDhbieFtw2YDEUYZ93yGEEJXDq_jjtxO9pRBsgv8) I modified it to use the math_0shot_gen_393424.py file with the following content: ![image](https://private-user-images.githubusercontent.com/76173133/377706318-bba2a73a-602c-476e-8cdf-1b8d43479219.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA5Njk1MzUsIm5iZiI6MTczMDk2OTIzNSwicGF0aCI6Ii83NjE3MzEzMy8zNzc3MDYzMTgtYmJhMmE3M2EtNjAyYy00NzZlLThjZGYtMWI4ZDQzNDc5MjE5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA3VDA4NDcxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI2ZDBhMjU0ZDQ0MWYxZDc1MmM0ZjAzMDQzMjRjMjBkMzA0MDVjNTEyODhjMjk0MzFkODQyZWJjYTU0YmQ0NGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.SQoanllLEbILpRTejuFyAZM7ybqn49P6jLs698Y5tu0) After stress testing, the score improved from 49.18 to 75.18.

hello，why I use the math_0shot_gen_393424.py file to test by OpenCompass just get 58.16 score (Qwen2.5-Math-72B-Instruct). And the model answer more other contents, do you meet this problem just like this:
![image](https://github.com/user-attachments/assets/5970bef7-32d1-46fb-8683-a5e47d3259f1)

**Comment by jklj077** [comment]
@13416157913 please raise that at https://github.com/QwenLM/Qwen2.5-Math
